src/extension-test.ts

```diff
@@ -0,0 +1,136 @@
+/**
+    Copyright 2023 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import {
+  RunnerExtension,
+  createRunner,
+  parse,
+  stringify,
+} from '../lib/main.js';
+import { importExtensionFromPath } from './CLIUtils.js';
+import http from 'http';
+import { files, recording, expectedLog } from './Spec.js';
+import assert from 'assert/strict';
+import { spawn } from 'node:child_process';
```

Comment:
If we import it like so we need to bump the min version to node `14.18.0` -> https://nodejs.org/api/modules.html#core-modules. As it will brake on older versions.